subprojects/plugins/src/integTest/groovy/org/gradle/java/compile/BasicJavaCompilerIntegrationSpec.groovy

```diff
@@ -144,6 +146,71 @@ public class FxApp extends Application {
         succeeds("compileJava")
     }
 
+    @Requires(adhoc = { AvailableJavaHomes.getJdk(JavaVersion.VERSION_11) != null })
+    def 'ignores project level compatibility when using toolchain'() {
+        given:
+        goodCode()
+        buildFile << """
+java.sourceCompatibility = JavaVersion.VERSION_14
+java.targetCompatibility = JavaVersion.VERSION_15
```

Comment:
Hm I'd like to better understand why we can't detect this. While we can't tell via the getter (because of the fallback), couldn't we introduce a get on `DefaultJavaPluginConvention` to get us the raw value (and thus non-null if it set -> warn user)

Reply:
It is not only the detection itself that is tricky, it is also about when we want to warn.
Ideally we would warn at configuration, but in practice we may only be able to warn at runtime, and it might be tricky to warn only once.

Reply:
I think its fair for the build to just fail in that case. I'd rather have a failing build than something that works but ignores my setup. Similar to how we fail the build when you try to use exectuable+toolchain support